### PR TITLE
Add provider for https://ok.ru

### PR DESCRIPTION
**Pull request by ajupov**
Added provider for Odnoklassniki (https://ok.ru).

### CODE REVIEW

**Comment by kevinchalet** [comment]
Thanks for your contribution. I'll take a look and merge it once 1.0 and 2.0 ship (next week).

**Comment by mokeev1995** [comment]
@PinpointTownes Hey, will it be added in 3.0 or some other release?

**Comment by Vertigo093i** [comment]
It is already available here: https://github.com/Digillect/AspNetCoreOAuthProviders.

File: src/AspNet.Security.OAuth.Odnoklassniki/OdnoklassnikiAuthenticationHandler.cs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            context.RunClaimActions(payload);
+
+            await Options.Events.CreatingTicket(context);
+
+            return new AuthenticationTicket(context.Principal, context.Properties, Scheme.Name);
+        }
+
+        private static string GetMd5([NotNull] string value)
+        {
+            var md5 = MD5.Create();
```
Comment (martincostello) [inline src/AspNet.Security.OAuth.Odnoklassniki/OdnoklassnikiAuthenticationHandler.cs:64]: `MD5` implements `IDisposable`, so this needs to be in a `using` block.

File: src/AspNet.Security.OAuth.Odnoklassniki/OdnoklassnikiAuthenticationHandler.cs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        {
+            return $"application_key={Options.ApplicationKey}" +
+                   "format=json" +
+                   "method=users.getCurrentUser" +
+                   $"{md5Inner}";
+        }
+
+        private string PrepareUserInfoAddress([NotNull] string accessToken, [NotNull] string sig)
+        {
+            var address = QueryHelpers.AddQueryString(Options.UserInformationEndpoint, "application_key", Options.ApplicationKey);
```
Comment (martincostello) [inline src/AspNet.Security.OAuth.Odnoklassniki/OdnoklassnikiAuthenticationHandler.cs:87]: Could use a dictionary like many of the other providers do and do this as a single operation?

**Comment by martincostello** [comment]
Thanks for the PR - as part of the preparation for ASP.NET Core 3.0 support, tests are being added to help make things easier to maintain and validate going forwards (see #292).

Once tests are merged into the dev branch, could you copy the approach to add tests for the new provider into this PR please?

**Comment by martincostello** [comment]
#280 has been merged to `dev` now. If you rebase and update any tests as appropriate, we can look at merging this PR soon.

**Comment by martincostello** [comment]
Closing due to inactivity and merge conflicts - please re-open and rebase if you would like to pursue adding a provider for ok.ru in a future release.

**Comment by martincostello** [comment]
Superseded by #311.

**Pull request closed without merging**